### DISCUSSION TITLE
Update certain column not by owner

### DISCUSSION THREAD

**Discussion by hilmanski** [supabase/supabase#2951] [Questions] [answered]
I want to create simple forum
Where each post has column updated_at 

This column will be updated each time someone reply on this post

The problem is: I want to use Row Level security, where “update” operation on this post only allowed by owner

But I also want to set a function/trigger, when anyone reply on this thread the updated_at column will automatically updated. 

Currently its not working since I protect my post with row level security where only owner can update it.

How would you guys tackle this situation?

**Comment by silentworks** [comment]
Using a function/trigger should bypass RLS from what I remember.

I think there is something wrong with your db design, in reality, the post was never updated, only a reply was created. If you want to get latest posts by latest replies then you could create a View with a query to return that data.

**Reply by hilmanski** [reply]
Yes, I agree with that, I think it's a hacky solution so I can filter by last updated at. My thoughts is replies is part of that post. Or maybe just rename it to last_activity_at. 

"Using a function/trigger should bypass RLS from what I remember."
I do really hope that's what happen. Unfortunately not, currently only when the owner of the post replying.. then the timestamp is updated. 

I will take a look at the View options. thanks

**Reply by w3b6x9** [reply]
Check out this resource on policies and security definer functions: https://supabase.io/docs/guides/auth#policies-with-security-definer-functions.

**Reply by hilmanski** [reply]
thanks @w3b6x9 will check it out. 

unrelated from link above.
I Create supabase client with service_role key as second parameter, I think will bypass RLS, but it doesn't, do you know why? I read in the docs. using service key role when creating supabase client should bypass the RLS

**Reply by hilmanski** [reply]
Hi @w3b6x9  Can you also move your answer on separate "asnwer" 
I want to mark it as the solution so others who have same problem with me can use it as reference

**Reply by hilmanski** [reply]
Can I also only allowed specific column to be updated by other people? I can see the risk of this, when people know the anon key, than people can update "everything"

**Reply by w3b6x9** [reply]
> using service key role when creating supabase client should bypass the RLS

this is correct! Please use caution when using the `service_role` API key to prevent its leak b/c it enables admin privileges.

> I think will bypass RLS, but it doesn't, do you know why?

hmm, this is odd. Can you please email support@supabase.io with additional information about your setup and we can investigate it further.

> Can I also only allowed specific column to be updated by other people?

I'm not sure what you mean so can you please elaborate? I think the straightforward answer is to properly set up RLS with the right function/trigger when using the anon key.

**Comment by w3b6x9** [comment, marked as answer]
@hilmanski

> Hi @w3b6x9 Can you also move your answer on separate "asnwer"
> I want to mark it as the solution so others who have same problem with me can use it as reference

https://github.com/supabase/supabase/discussions/2951#discussioncomment-1236797

**Reply by hilmanski** [reply]
thanks for you help @w3b6x9
